### ISSUE TITLE
Explanation: Project Architecture

### ISSUE THREAD

**Issue by renilbabu03** [BlackBoxVision/typescript-hapi-starter#31]
It would be really nice if you could explain the project architecture. Like explaining the following:
1. Resolve file
2. Repository file
3. Controller file (not necessary)

I don't understand the use of a resolve file. Looks like you are using a Architectural Pattern. Please explain to us which pattern and a general flow.

**Comment by laurilarjo** [comment]
Yes we could add some description of that architecture. The resolver decouples the repository from the controller, but it might make it harder to understand the project. 

You can also use this as a base by ignoring the common base-controller, base-repository and base-resolver and just writing your own methods to the user-controller. That's how I started to use this project, as I didn't have any database, but instead I was proxying and transforming data from another backend.

Could you tell us your use case why you're looking at this starter kit, and what your needs are? That would also give insight on how to improve this.
